### DISCUSSION TITLE
Poetry does not cache files and downloads them multiple times

### DISCUSSION THREAD

**Discussion by BlueskyFR** [python-poetry/poetry#7791] [Q&A] [answered]
When running:

```bash
poetry add https://download.pytorch.org/whl/cu118/torch-2.0.0%2Bcu118-cp310-cp310-linux_x86_64.whl
```

(which is a big file), Poetry waits a really long time, then downloads the file, then generates the lockfile, downloads the dependencies **including the same file again**, before finally completing.

Why is that?
I also feel like Poetry is unable to see that `torch==2.0.0+cu118` is the same as `torch==2.0.0` while pip does

**Comment by radoering** [comment, marked as answer]
#2415
